src/solvers/dgmulti/dg.jl

```diff
@@ -195,6 +199,37 @@ function calc_interface_flux!(cache, surface_integral::SurfaceIntegralWeakForm,
   end
 end
 
+function calc_interface_flux!(cache, surface_integral::SurfaceIntegralWeakForm,
+                              mesh::VertexMappedMesh,
+                              have_nonconservative_terms::Val{true}, equations,
+                              dg::DGMulti{NDIMS}) where {NDIMS}
+
+  flux_conservative, flux_nonconservative = surface_integral.surface_flux
+  md = mesh.md
+  @unpack mapM, mapP, nxyzJ, Jf = md
+  @unpack u_face_values, flux_face_values = cache
+
+  @threaded for face_node_index in each_face_node_global(mesh, dg, cache)
+
+    # inner (idM -> minus) and outer (idP -> plus) indices
+    idM, idP = mapM[face_node_index], mapP[face_node_index]
+    uM = u_face_values[idM]
+
+    # compute flux if node is not a boundary node
+    if idM != idP
+      uP = u_face_values[idP]
+      normal = SVector{NDIMS}(getindex.(nxyzJ, idM)) / Jf[idM]
+      conservative_part = flux_conservative(uM, uP, normal, equations)
+
+      # the second "normal" argument is normal_vector_average?
+      nonconservative_part = flux_nonconservative(uM, uP, normal, normal, equations)
+      # The factor 0.5 is necessary for the nonconservative fluxes based on the
+      # interpretation of global SBP operators.
+      flux_face_values[idM] = (conservative_part + .5 * nonconservative_part) * Jf[idM]
```

Comment:
I think the double-computation is an artifact of looping over interfaces rather than elements. For loops over elements, there is a clearer distinction between`u_mine` and `u_other`. 

However, I'm not 100% sure this is correct. Any good way to test this?

Reply:
Moved to #860

Reply:
Ah, I see. So `each_face_node_global(mesh, dg, cache)` visits each interface node twice (once from each adjacent element)? If so, it would be great if you could add a corresponding comment before the loops `@threaded for face_node_index in each_face_node_global(mesh, dg, cache)` and an additional comment here, e.g., 
```julia
      nonconservative_part = flux_nonconservative(uM, uP, normal, normal, equations)
      # The factor 0.5 is necessary for the nonconservative fluxes based on the
      # interpretation of global SBP operators.
      # In contrast to other mesh types, only one nonconservative part needs to be 
      # computed since we loop over the elements, not the interfaces.
      flux_face_values[idM] = (conservative_part + .5 * nonconservative_part) * Jf[idM]
```

Reply:
> However, I'm not 100% sure this is correct. Any good way to test this?

The fact that it's not blowing up is already a good test :sweat_smile: Otherwise, you could check a nonconservative linear advection equation with piecewise constant approximations (such that you get only interface terms), similar to [our tutorial](https://trixi-framework.github.io/Trixi.jl/stable/adding_new_equations/nonconservative_advection/). But that's not necessary from my point of view if you're looping over elements - then, this approach looks good to me.